trinity/plugins/builtin/light_peer_chain_bridge/light_peer_chain_bridge.py

```diff
@@ -0,0 +1,114 @@
+import asyncio
+from enum import (
+    auto,
+    Enum,
+)
+from typing import (
+    Any,
+)
+from eth_typing import (
+    Hash32,
+)
+
+from eth.rlp.headers import (
+    BlockHeader,
+)
+from lahja import (
+    BaseEvent,
+    Endpoint,
+)
+from trinity.rlp.block_body import BlockBody
+from trinity.sync.light.service import (
+    LightPeerChain,
+)
+
+
+# This is an ad-hoc PoC attempt. I'm sure we can use some meta programming magic
+# to make this whole thing effectively become zero maintanance (e.g. new / changed
+# APIs in LightPeerChain are automatically accessible through event-bus calls)
+class LightPeerChainAPI:
+
+    GET_BLOCK_HEADER_BY_HASH = 'get_block_header_by_hash'
+    GET_BLOCK_BODY_BY_HASH = 'get_block_body_by_hash'
+    GET_FOOBAR = 'get_foobar'
+
+class LightPeerChainRequest(BaseEvent):
+
+    def __init__(self, api: LightPeerChainAPI, payload: Any = None):
+        self.api = api
+        self.payload = payload
+
+class LightPeerChainResponse(BaseEvent):
+
+    def __init__(self, payload: Any = None):
+        self.payload = payload
+
+
+
+class LightPeerChainEventBusBridge:
+
+    def __init__(self, chain: LightPeerChain, event_bus: Endpoint):
+        self.chain = chain
+        self.event_bus = event_bus
+        asyncio.ensure_future(self.answer_requests())
+
+    async def answer_requests(self) -> None:
+        async for event in self.event_bus.stream(LightPeerChainRequest):
+
+            method = getattr(self.chain, event.api)
+
+            if not callable(method):
+                self.event_bus.broadcast(
+                    LightPeerChainResponse(Exception(f"Not a method: {event.api}")), event.broadcast_config()
+                )
+                continue
+
+            try:
+                response = await method(*event.payload)
+            except Exception as e:
+                # we send the exception over to raise it on the other end
+                self.event_bus.broadcast(LightPeerChainResponse(e), event.broadcast_config())
+            else:
+                self.event_bus.broadcast(LightPeerChainResponse(response), event.broadcast_config())
+
+
+class EventBusLightPeerChain:
```

Comment:
I don't know if it actually makes sense, but looks like inheriting this from `BaseChain` the same way that `LightDispatchChain` does would provide some glue to ensure that the types and return types match up.

Reply:
Yes, I think so, too. Will comment a bit more on that further down.